### DISCUSSION TITLE
Filter individual values of repeater/flexible fields (and other nested fields)

### DISCUSSION THREAD

**Discussion by furai** [wp-graphql/wpgraphql-acf#196] [Q&A]
Hey @jasonbahl,

I have an issue and I can't seem to be able to solve it. In older implementation of `wp-graqphql-acf` I was able to use `graphql_acf_field_value` filter to change values of subfields in reapeater/flexible_fields (and other types that have nested values that I'm forgetting now). 

I had a piece of code that was checking the type of the field and if it was `text` or `textarea` then I was applying `do_shortcode` function to its value. 

This still works as expected with new filter called `wpgraphql/acf/field_value` for simple fields but I can't seem to be able to do the same for individual values in a reapeater/flexible_content. 

Is there any other filter to do just that? I've browsed the code and couldn't find anything useful. Seems like right now I'm getting repeaters/flexible_fields as one big blog of data with values in 2D array.

What I'm after is to be able to filter these subfields the same as parent field. Could you help me with that?

Cheers,
Furai

**Comment by furai** [comment]
So I've rethought whole approach and instead of using wpgraphql-acf filter I've used `acf/format_value` filter from ACF Pro plugin.

So only request would be to add some filter to be able to change the `should_format_field_value` function result.

EDIT: After trying the code out I've ended up with using combination of both filters - `wpgraphql/acf/field_value` for subfields and `acf/format_value` for parent fields.

**Reply by furai** [reply]
Hmm, I'm still having issue though with adding custom field next to each wysiwyg in acf. I had a wysiwyg acf field and next to each of those I've added parsedWysiwyg that was doing some logic to change wysiwyg to json.  

Now I'm really stumped as how to do it. @jasonbahl any ideas?

**Reply by jasonbahl** [reply]
@furai it looks like this might be related to this issue? https://github.com/wp-graphql/wpgraphql-acf/issues/149

I have not had a chance to dig into this further.

**Reply by furai** [reply]
Thanks for taking time to answer it. I've solved my issue with wysiwyg by using `wpgraphql/acf/get_graphql_field_config` filter.
Now I'm facing another issue where I need to port my custom solution for my custom acf multisite post object. I need the connection to resolve in context of another blog. I'm kind of fighting with it but I have some ideas. Analysing Defer in xdebug and the flow of wpgraphql plugin proves tiresome at times.

**Reply by furai** [reply]
I'm not sure how to best use these discussions...so sorry if this gets spammy. Anyway another thing I've noticed is that HierarchicalTermNode returns just the parentid and I need a whole parent node. Is there any way to get that? @jasonbahl Do I need to override something?

**Reply by jasonbahl** [reply]
> I'm not sure how to best use these discussions

@furai tbh, I don't know the best way to use these discussions either 🤣 However we use them, though, the information now becomes searchable and hopefully can help someone else in the future. 

> HierarchicalTermNode returns just the parentid and I need a whole parent node. Is there any way to get that?

The Interface `HierarchicalTermNode` does not have a `parent` connection defined, but the individual Types (i.e. Category) do. 

I believe there's some context on the technical limitations in the conversation in this PR (https://github.com/wp-graphql/wp-graphql/pull/2026)

So for now, I think you could do something along these lines, where you ask for the parent on the specific Object Type (i.e. `Category`): 

```graphql
query GetCategories {
  terms(first:100) {
    nodes {
      id
      ...on Category {
        parent {
          node {
            __typename
            id
            name
          }
        }
      }
    }
  }
}
```

**Reply by furai** [reply]
@jasonbahl  Thanks, that did the trick! Now I see that some of our queries were using children selector...I'll check if I can fix it on my own but if you have any suggestions then please do share them! I really appreciate your help and thank you for your precious time. 

EDIT: it seems children works the same way as parent so that's solved.
EDIT2: Also I had another issue with `contentNodes` that this solved as well.

**Reply by furai** [reply]
@jasonbahl so once I was done with all the problems on the backend and on frontend with adjusting queries to new data format - I'm now really struggling with typescript. We use codegen with schema from WP generated using introspection to get types into our project. Now complexity of getting correct data from nodes increased greatly.

Before most of the nodes were of concrete types - now they can be arrays of literally every CPT we have in the system. Do you have any tips on how to improve on that? Do you have any experience with typescript? Maybe `... on Category` and such can be changed to something else in queries?

**Reply by jasonbahl** [reply]
@furai I'm not too familiar with codegen for generating Types so I can't give advice or direction there.

But when querying an Interface (i.e. the `terms` field), it is possible to get any number of Possible Types in response, so the consumer needs to be prepared to get different Types of terms, hence why different Types are being generated.

If you only want Categories instead of terms of any type, you could query for:

```
query GetCategories {
  categories(first:100) { <-- instead of "terms"
    nodes {
      id
      parent {
        node {
          __typename
          id
          name
        }
      }
    }
  }
}
```

**Reply by furai** [reply]
@jasonbahl another thing I've noticed right now that wysiwyg seems to have weird encoding issues when encountering shortcode. Like output is different than what it was. Like something was done before doing shortcodes. Seems like something is adding newlines/encoding content. Or maybe after doing shortcodes.

The same thing in normal content in post is completely different in wysiwyg field when it really shouldn't be the case.
Also seems that when fetching the field directly through acf functions all is ok. 
I think shortcode has to be in the opening and closing for it to happen.

Maybe the_content filter is applied multiple times? Though I'm not sure how that would work.

EDIT: Seems that formatting "flexible_content", "wysiwyg" and "repeater" does that in should format. Any idea how to fix that?

EDIT2: I'm not sure but in wp-graphql plugin applys the_content filter only to rendered content. Maybe that's something to look into with this WYSIWYG issue?

**Reply by furai** [reply]
@jasonbahl any idea about the wysiwyg issue above? I think we need badly that should_format filter.

**Reply by furai** [reply]
I've ended up just disabling formatting for any wysiwyg fields in via hook in ACF plugin itself.
